source/common/common/base64.h

```diff
@@ -12,11 +12,28 @@ class Base64 {
   static std::string encode(const Buffer::Instance& buffer, uint64_t length);
 
   /**
+   * Base64 encode an input char buffer with a given length.
+   * @param input char array to encode.
+   * @param length of the input array.
+   */
+  static std::string encode(const char* input, uint64_t length);
+
+  /**
    * Base64 decode an input string.
    * @param input supplies the input to decode.
    *
    * Note, decoded string may contain '\0' at any position, it should be treated as a sequence of
    * bytes.
    */
   static std::string decode(const std::string& input);
+
+private:
+  /**
+   * Helper methods for encoding.
+   */
+  static void encodeBase(const uint8_t cur_char, uint64_t pos, uint8_t& next_c, std::string& ret);
```

Comment:
nit: What does this method do? (you have comment for encodeLast, but none for encodeBase)